### PR TITLE
Fix match return vertex undefined labels 

### PR DESCRIPTION
**Pull request by Aiee**
Address a problem that storage does not recognize vertex tag without property

Fix https://jira.nebula-graph.io/browse/NG-344

Updates:
1. add a new interface in metaClient getAllLatestTagSchema()
2. update tck frame to add support for using Examples keyword when loading data into space
3. replace getAllVerTagSchema() with getAllLatestVerTagSchema() in graph

Depends on https://github.com/vesoft-inc/nebula-storage/pull/288 and https://github.com/vesoft-inc/nebula-common/pull/379

### CODE REVIEW

File: src/parser/parser.yy [hunk 1]
```diff
@@ -827,6 +827,9 @@ function_call_expression
     | KW_DATETIME L_PAREN opt_argument_list R_PAREN {
         $$ = new FunctionCallExpression(new std::string("datetime"), $3);
     }
+    | KW_TAGS L_PAREN opt_argument_list R_PAREN {
+        $$ = new FunctionCallExpression(new std::string("tags"), $3);
+    }
```
Comment (CPWstatic) [inline src/parser/parser.yy:832]: Why we had this function a standalone rule?

Comment (yixinglu) [reply src/parser/parser.yy:832]: support to use `tags(v)` in expression directly.

File: src/planner/match/MatchClausePlanner.cpp [hunk 2]
```diff
@@ -297,6 +304,35 @@ Status MatchClausePlanner::appendFetchVertexPlan(const Expression* nodeFilter,
     return Status::OK();
 }
 
+StatusOr<std::vector<storage::cpp2::VertexProp>>
+        MatchClausePlanner::flattenTags(QueryContext *qctx, const SpaceInfo& space) {
```
Comment (yixinglu) [inline src/planner/match/MatchClausePlanner.cpp:308]: please format your changes

File: src/planner/match/MatchClausePlanner.h [hunk 3]
```diff
@@ -80,6 +80,9 @@ class MatchClausePlanner final : public CypherClausePlanner {
                                  size_t nodeInfoSize) const;
 
     Status appendFilterPlan(MatchClauseContext* matchClauseCtx, SubPlan& subplan);
+    // Fetch all tags in the space and retrieve props from tags
+    StatusOr<std::vector<storage::cpp2::VertexProp>>
+        flattenTags(QueryContext *qctx, const SpaceInfo& space);
```
Comment (yixinglu) [inline src/planner/match/MatchClausePlanner.h:85]: ditto

File: tests/tck/features/bugfix/MatchReturnEmptyTag.feature [hunk 4]
```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2020 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
+@test
```
Comment (yixinglu) [inline tests/tck/features/bugfix/MatchReturnEmptyTag.feature:5]: delete this `test` mark

**Review by yixinglu — DISMISSED** [review]

**Review by jievince — DISMISSED** [review]

File: src/context/Iterator.cpp [hunk 5]
```diff
@@ -549,7 +551,11 @@ Value PropIter::getVertex() const {
         Tag tag;
         tag.name = tagProp.first;
         for (auto& propIndex : tagProp.second) {
-            tag.props.emplace(propIndex.first, row[propIndex.second]);
+            if (propIndex.first == "_tag") {
```
Comment (laura-ding) [inline src/context/Iterator.cpp:554]: use `kTag`

File: tests/tck/features/bugfix/MatchReturnEmptyTag.feature [hunk 6]
```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2020 vesoft inc. All rights reserved.
```
Comment (Shylock-Hg) [inline tests/tck/features/bugfix/MatchReturnEmptyTag.feature:1]: Add int vid cases.

File: src/planner/match/MatchClausePlanner.cpp [hunk 7]
```diff
@@ -297,6 +304,35 @@ Status MatchClausePlanner::appendFetchVertexPlan(const Expression* nodeFilter,
     return Status::OK();
 }
 
+StatusOr<std::vector<storage::cpp2::VertexProp>>
+MatchClausePlanner::flattenTags(QueryContext *qctx, const SpaceInfo& space) {
+    // Get all tags in the space
+    const auto allTagsResult = qctx->schemaMng()->getAllVerTagSchema(space.id);
```
Comment (laura-ding) [inline src/planner/match/MatchClausePlanner.cpp:310]: the `getAllVerTagSchema` will get all version tags, and the result will include different version tags, I think you need to get the lastest tag, not all version tags.

File: tests/tck/features/bugfix/MatchReturnEmptyTag.feature (hunk 6 above)
Comment (Aiee) [reply tests/tck/features/bugfix/MatchReturnEmptyTag.feature:1]: Tests were added

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Pull request merged**